### PR TITLE
[JENKINS-43802] Implementing retrieve/retrieveRevisions with an Item context

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/scm-api-plugin/pull/65.

### CODE REVIEW

**Comment by bitwiseman** [comment]
Waiting on upstream.

File: pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
         <changelist>-SNAPSHOT</changelist>
         <java.level>8</java.level>
         <jenkins.version>2.121.3</jenkins.version>
-        <scm-api.version>2.3.0</scm-api.version>
+        <scm-api.version>2.5.0-rc464.915933672136</scm-api.version> <!-- TODO https://github.com/jenkinsci/scm-api-plugin/pull/65 -->
```
Comment (bitwiseman) [inline pom.xml]:
```suggestion
        <scm-api-plugin.version>2.6.2</scm-api-plugin.version>
```

**Comment by jglick** [comment]
```
java.io.IOException: SCM API Plugin v2.6.2 failed to load.
 - You must update Jenkins from v2.121.3 to v2.164.3 or later to run this plugin.
```

File: pom.xml [hunk 2]
```diff
@@ -25,7 +25,7 @@
         <changelist>-SNAPSHOT</changelist>
         <java.level>8</java.level>
         <jenkins.version>2.121.3</jenkins.version>
-        <scm-api.version>2.3.0</scm-api.version>
+        <scm-api.version>2.6.2</scm-api.version>
```
Comment (bitwiseman) [inline pom.xml]:
```suggestion
        <scm-api.version>2.6.3</scm-api.version>
```

**Pull request merged**